exercises/algorithms/src/test/java/com/cbfacademy/BinarySearchTest.java

```diff
@@ -0,0 +1,77 @@
+package com.cbfacademy;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.Arrays;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import com.cbfacademy.search.BinarySearch;
+import com.cbfacademy.search.Search;
+
+@DisplayName(value = "BinarySearch Exercise")
+public class BinarySearchTest {
+    Search binarySearch;
+    int[] testArray = new int[50];
+
+    @BeforeEach
+    public void setUp() {
+        binarySearch = new BinarySearch();
+        Arrays.setAll(testArray, position -> position + 1);
+    }
+
+    @Test
+    @DisplayName("returns this unit's name")
+    public void returnUnitName() {
+        String name = binarySearch.getName();
+
+        assertThat(name, is("Binary Search Algorithm Exercise"));
+    }
+
+    @Test
+    @DisplayName("search returns the index of the target value when it is in the middle of the array")
+    public void returnsIndexWhenTargetIsMidPointInArray() {
```

Comment:
Just as a suggestion, prefixing the test name with the method name under test makes it easier to group and find tests for a specific method, e.g.

```java
public void search_ReturnsIndexWhenTargetIsMidPointInArray()
```